carmm/meshgrid/meshgrid_void.py

```diff
@@ -0,0 +1,140 @@
+def void_find(Ucell, atom, mic=True, coarseness=1):
+    """
+    Defines the void centres and radii of points across the meshgrid. Coarseness factor reduces number of probe points
+    by skipping over points in the grid. Uses 99.99 as a junk value.
+    Args:
+        Ucell: unit_cell object
+            Contains meshgrid and associated unit cell information.
+        atom: Atoms object
+            Supplies coordinates and atomic information.
+        mic: logical
+            Use minimum image convention.
+        coarseness: integer
+            Skips over an integer number of points
+    Return:
+        void_centres: numpy array
+            Atomic coordiantes of the void centres.
+        void_radii: numpy array
+            Radius of the void centres
+    """
+
+    import numpy as np
+    from ase.data.vdw_alvarez import vdw_radii
+    from ase.data import atomic_numbers
+
+    void_centres = []
+    void_radii = []
+    atom_radii = []
+
+    for atom_co in range(len(atom.positions)):
+        at_symbols = atom.get_chemical_symbols()
+        at_n = atomic_numbers[at_symbols[atom_co]]
+        atom_radius = vdw_radii[at_n]
+        atom_radii.append(atom_radius)
+
+    atom_radii = np.array([atom_radii]).flatten()
+
+    for i in range(0, Ucell.nx, coarseness):
+        for j in range(0, Ucell.ny, coarseness):
+            for k in range(0, Ucell.nz, coarseness):
```

Comment:
I would use `itertools.product` or `np.ndindex` here. But is the matter of taste.

Reply:
Will bear this in mind for future, but for the sake of time will leave it as is in this code.